### ISSUE TITLE
I get an error when i convert to caffe using tools offered by the MMdnn project

### ISSUE THREAD

**Issue by 791188299** [tobybreckon/fire-detection-cnn#15]
I have got three .pb files .  When i convert to caffe modes using tools offered by the MMdnn project ,this happens:
ValueError: NodeDef expected inputs '' do not match 1 inputs specified; Op<name=Const; signature= -> output:dtype; attr=value:tensor; attr=dtype:type>; NodeDef: {{node Dropout/cond/dropout/random_uniform/max}} = Const[dtype=DT_FLOAT, value=Tensor<type: float shape: [] values: 1>](Dropout/cond/Switch:1)

Please give me a hand

**Comment by tobybreckon** [comment]
Do you get this with firenet only, or all three of them? (Only firenet uses dropout).

If it is firenet only, can you try the following:
-- comment out lines network = dropout (...
-- rerun test on video file (does it still work correctly? , Should do, please check)
-- rerun conversion to .pb and subsequent opencv dnn test (still working, yes/no?)
-- retry conversion with MMdnn (solved?)

Please post back your findings, it would be very helpful to know.

**Comment by 791188299** [comment]
> Do you get this with firenet only, or all three of them? (Only firenet uses dropout).
> 
> If it is firenet only, can you try the following:
> -- comment out lines network = dropout (...
> -- rerun test on video file (does it still work correctly? , Should do, please check)
> -- rerun conversion to .pb and subsequent opencv dnn test (still working, yes/no?)
> -- retry conversion with MMdnn (solved?)
> 
> Please post back your findings, it would be very helpful to know.
Thank you for your reply! 
I get this with all three of them. When I using  tool summarize_graph of tensorflow with .pb files,  it gets  
"--input_layer=InputData/X 
--input_layer_type=float 
--input_layer_shape= 
--output_layer=FullyConnected/Softmax" 
 why its  input_layer_shape is nothing

**Comment by 791188299** [comment]
Now I try to test the steps above.

**Comment by tobybreckon** [comment]
@791188299 - did this work?

**Comment by 791188299** [comment]
> @791188299 - did this work?
remove dropout layer  it still work correctly
rerun conversion to .pb file no problem
retry conversion with MMdnn ,it gets  an errors about LRN layer parameter when makeing net . 
ignore that parameter, successful convert to caffe .the test results with firenet are correct.

**Comment by tobybreckon** [comment]
Thanks - can you summarize the code changes needed to the network creation functions to this to work?

Is it ...
- comment out all dropout() lines?
- remove ... from all LRN lines?

**Comment by 791188299** [comment]
1)ValueError: NodeDef expected inputs '' do not match 1 inputs specified; Op<name=Const; signature= -> output:dtype; attr=value:tensor; attr=dtype:type>; NodeDef: {{node Dropout/cond/dropout/random_uniform/max}} = Const[dtype=DT_FLOAT, value=Tensor<type: float shape: [] values: 1>](Dropout/cond/Switch:1)

comment out all dropout() line.(fienet :#network = dropout(network, 0.5); inceptionv1onfire:#pool5_7_7 = dropout(pool5_7_7, 0.4))

2)File "......./caffe/python/caffe/net_spec.py", line 79, in assign_proto
    setattr(proto, name, val)
  TypeError: None has type NoneType, but expected one of: int, long, float

File "......./caffe/python/caffe/net_spec.py"
 Before modification: 
            for k, v in six.iteritems(self.params):
                # special case to handle generic *params
                if k.endswith('param'):
                    assign_proto(layer, k, v)
After modification:
            for k, v in six.iteritems(self.params):
                if layer.name == "LocalResponseNormalization_LocalResponseNormalization" and k == "k":
                    continue
                # special case to handle generic *params
                if k.endswith('param'):
                    assign_proto(layer, k, v)

**Comment by 791188299** [comment]
Thanks for your hlep, by the way, is the result of pre-trained model consistent with the model in the paper?

**Comment by tobybreckon** [comment]
Yes - the are the models the paper results were generated from.

However, with the removal of dropout and LRN at inference once you convert to Caffe, performance may differ slightly.

I think I need to add a regression test into the repo to allow us to confirm the effect of changes.

**Comment by 791188299** [comment]
OK,Thank you very much!
